docs/design/ROADMAP.md

```diff
@@ -0,0 +1,56 @@
+## Introduction
+
+This document serves as a high-level reference for Envoy Gateway users and contributors to understand the direction of
+the project.
+
+## Contributing to the Roadmap
+
+- To propose a roadmap item, create an [issue][issue] or join a [community meeting][meeting] to discuss your desired
+  feature. Update this document if your requested feature is accepted for a particular [milestone][milestones].
+- To help with an existing roadmap item, comment on the associated issue. If an item doesn't have an issue, create one
```

Comment:
I really like how the specific release Roadmap shares a high level direction of the release, some specific feature support and links to the milestone which has all the exact GH issues/details.
I prefer if users didn't need to keep raising PRs to update ROADMAP.md when their issues got linked to a specific release, just tagging the GH issue to a specific milestone should be adequate imho